### ISSUE TITLE
Annotated cannot wrap ParamSpec args/kwargs

### ISSUE THREAD

**Issue by GBeauregard** [microsoft/pyright#2995]
**Describe the bug**
Annotated cannot wrap paramspec args and kwargs

**To Reproduce**
```python
import logging
from typing import Annotated, Callable, ParamSpec, TypeVar

T = TypeVar("T")
P = ParamSpec("P")

def add_logging(f: Callable[P, T]) -> Callable[P, T]:
    # ParamSpec not allowed in this context
    def inner(*args: Annotated[P.args, "meta"], **kwargs: P.kwargs) -> T:
        logging.info(f"{f.__name__} was called")
        return f(*args, **kwargs)
    return inner

@add_logging
def add_two(x: float, y: float) -> float:
    return x + y
```

**Expected behavior**
pyright should allow this. I think the Annotated PEP itself is clear enough this shouldn't be behaving differently.

**VS Code extension or command-line**
commandline 1.1.218

**Additional context**
This is *not* allowed at runtime, but it seems clearly a bug. I filed a bug at [bpo-46643](https://bugs.python.org/issue46643) and I intend to either fix it there or with [bpo-46644](https://bugs.python.org/issue46644). Maybe both.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by GBeauregard** [comment]
This is inane so I'm not going to make an issue, but while playing with [bpo-46655](https://bugs.python.org/issue46655) I discovered pyright doesn't allow stringified `TypeAlias`. Jelle mentioned `Required` and `NotRequired` as well. We both agree stringifying these is pretty pointless, but I wanted to do my due diligence to point it out.

**Comment by erictraut** [comment]
Yes, this is as designed.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.219, which I just published. It will also be in the next release of pylance.
